dts/arm/unisoc/uwp5661.dtsi

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2018, UNISOC Incorporated
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <mem.h>
+
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m4";
+			reg = <0>;
+		};
+	};
+
+	sram0: memory@100000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0x100000 DT_SIZE_K(832)>;
+	};
+
+	flash0: flash@200000 {
+		compatible = "serial-flash";
+		reg = <0x2000000 DT_SIZE_M(4)>;
+		label = "UWP_FLASH";
+	};
+
+	soc {
+		intc: core_intc@40000000 {
+			compatible = "unisoc,uwp-intc";
+			reg = <0x40000000 0x40>;
+			interrupt-controller;
+			#interrupt-cells = <2>;
+			interrupts = <1 0>, <0 0>;
+		};
+
+		aon_intc: aon_intc@40800000 {
+			compatible = "unisoc,uwp-intc";
+			reg = <0x40800000 0x40>;
+			interrupt-controller;
+			#interrupt-cells = <2>;
+			interrupts = <20 0>;
+		};
+
+		uart0: uart@40038000 {
+			compatible = "unisoc,uwp-uart";
+			reg = <0x40038000 0x30>;
+			interrupts = <35 3>;
+			current-speed = <115200>;
+			clock-frequency = <26000000>;
+			label = "UART_0";
+			status = "disabled";
+		};
+
+		uart1: uart@40040000 {
+			compatible = "unisoc,uwp-uart";
+			reg = <0x40040000 0x30>;
+			interrupts = <36 3>;
+			current-speed = <115200>;
+			clock-frequency = <26000000>;
+			label = "UART_1";
+			status = "disabled";
+		};
+
+		uart2: uart@40838000 {
+			compatible = "unisoc,uwp-uart";
+			reg = <0x40838000 0x30>;
+			interrupts = <0x0D14 3>;
+			current-speed = <115200>;
+			clock-frequency = <26000000>;
+			label = "UART_2";
+			interrupt-parent = <&intc>;
+			status = "disabled";
+		};
+
+		wdog: watchdog@40001000 {
+			compatible = "unisoc,uwp-watchdog";
+			reg = <0x40001000 0x30>;
+			interrupts = <40 0>;
+			label = "UWP_WDT";
+		};
+
+		port0: gpio@40804000 {
+			compatible = "unisoc,uwp-gpio";
+			reg = <0x40804000 0x30>;
+			interrupts = <0xB14 0>;
```

Comment:
seems like interrupt-parent needs to be set.

Reply:
Yes, it is a 2nd-level interrupt for gpio.

Reply:
In addition, the detailed SoC documents include interrupt and register descriptions is been reviewing internally, it will be released very soon.